internal/services/webhooks/webhooks.go

```diff
@@ -0,0 +1,342 @@
+					if err != nil {
+						c.sc.Logger.Error().Err(err).Msgf("failed to encrypt access token: %s", err.Error())
+						return
+					}
+
+					encTokStr := base64.StdEncoding.EncodeToString(encTok)
+
+					_, err = c.sc.EngineRepository.WebhookWorker().UpsertWebhookWorker(context.Background(), &repository.UpsertWebhookWorkerOpts{
+						Name:       ww.Name,
+						URL:        ww.Url,
+						Secret:     ww.Secret,
+						TenantId:   tenantId,
+						TokenID:    &tok.TokenId,
+						TokenValue: &encTokStr,
+					})
+					if err != nil {
+						c.sc.Logger.Error().Err(err).Msgf("could not update webhook worker %s of tenant %s", id, tenantId)
+						return
+					}
+
+					token = tok.Token
+				}
+
+				cleanup, err := c.run(tenantId, ww, token, h)
+				if err != nil {
+					c.sc.Logger.Error().Err(err).Msgf("error running webhook worker %s of tenant %s healthcheck", id, tenantId)
+					return
+				}
+				if cleanup != nil {
+					c.cleanups[id] = cleanup
+				}
+			}()
+		}
+	}
+
+	return nil
+}
+
+type HealthCheckResponse struct {
+	Actions   []string `json:"actions"`
+	Workflows []string `json:"workflows"`
+}
+
+func (c *WebhooksController) healthcheck(ww *dbsqlc.WebhookWorker) (*HealthCheckResponse, error) {
+	secret, err := c.sc.Encryption.DecryptString(ww.Secret, sqlchelpers.UUIDToStr(ww.TenantId))
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := whrequest.Send(context.Background(), ww.Url, secret, struct {
+		Time time.Time `json:"time"`
+	}{
+		Time: time.Now(),
+	}, func(req *http.Request) {
+		req.Header.Set("X-Healthcheck", "true")
+	})
+	if err != nil {
+		return nil, fmt.Errorf("healthcheck request: %w", err)
+	}
+
+	var res HealthCheckResponse
+	err = json.Unmarshal(resp, &res)
+	if err != nil {
+		return nil, fmt.Errorf("could not unmarshal response body: %w", err)
+	}
+
+	return &res, nil
+}
+
+func (c *WebhooksController) run(tenantId string, webhookWorker *dbsqlc.WebhookWorker, token string, h *HealthCheckResponse) (func() error, error) {
+	id := sqlchelpers.UUIDToStr(webhookWorker.ID)
+
+	secret, err := c.sc.Encryption.DecryptString(webhookWorker.Secret, sqlchelpers.UUIDToStr(webhookWorker.TenantId))
+	if err != nil {
+		return nil, fmt.Errorf("could not decrypt webhook secret: %w", err)
+	}
+
+	ww, err := webhook.New(webhook.WorkerOpts{
+		Token:     token,
+		ID:        id,
+		Secret:    secret,
+		URL:       webhookWorker.Url,
+		Name:      webhookWorker.Name,
+		TenantID:  tenantId,
+		Actions:   h.Actions,
+		Workflows: h.Workflows,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not create webhook worker: %w", err)
+	}
+
+	var cleanups []func() error
+
+	cleanup, err := ww.Start()
+	if err != nil {
+		return nil, fmt.Errorf("could not start webhook worker: %w", err)
+	}
+
+	cleanups = append(cleanups, cleanup)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go func() {
+		ticker := time.NewTicker(5 * time.Second)
```

Comment:
less frequent? maybe once per minute?

at some point it would be nice to "invalidate" on code changes

Reply:
good catch I decreased it for testing

Reply:
Whenever it runs this it compares the hash of the actions and workflows, so it kinda invalidates on code changes if that's what you meant

Reply:
nice, i meant more for adding actions but i dont think its critical